src/ports/postgres/modules/deep_learning/madlib_keras.py_in

```diff
@@ -403,49 +408,84 @@ def should_compute_metrics_this_iter(curr_iter, metrics_compute_frequency,
     return (curr_iter)%metrics_compute_frequency == 0 or \
            curr_iter == num_iterations
 
+
 def fit_transition(state, dependent_var, independent_var, model_architecture,
                    compile_params, fit_params, current_seg_id, seg_ids,
                    images_per_seg, gpus_per_host, segments_per_host,
-                   prev_serialized_weights, **kwargs):
+                   prev_serialized_weights, is_final=True,
```

Comment:
Because these functions are also used in madlib_keras_fit and there we didn't modify anything. So these function default to single model data-parallel training via model averaging.